test/codegen-generated-classes-test/src/test/java/software/amazon/awssdk/services/metrics/AsyncCoreMetricsTest.java

```diff
@@ -113,46 +119,172 @@ public void apiCall_publisherOverriddenOnRequest_requestPublisherTakesPrecedence
     @Test
     public void apiCall_operationSuccessful_addsMetrics() {
         stubSuccessfulResponse();
+
         client.allTypes().join();
 
         ArgumentCaptor<MetricCollection> collectionCaptor = ArgumentCaptor.forClass(MetricCollection.class);
         verify(mockPublisher).publish(collectionCaptor.capture());
-
         MetricCollection capturedCollection = collectionCaptor.getValue();
 
-        assertThat(capturedCollection.name()).isEqualTo("ApiCall");
-        assertThat(capturedCollection.metricValues(CoreMetric.SERVICE_ID))
-            .containsExactly(SERVICE_ID);
-        assertThat(capturedCollection.metricValues(CoreMetric.OPERATION_NAME))
-            .containsExactly("AllTypes");
+        verifyApiCallCollection(capturedCollection);
+
+        assertThat(capturedCollection.children()).hasSize(1);
+        MetricCollection attemptCollection = capturedCollection.children().get(0);
+
+        assertThat(attemptCollection.name()).isEqualTo("ApiCallAttempt");
+        assertThat(attemptCollection.children()).isEmpty();
+
+        verifySuccessfulApiCallAttemptCollection(attemptCollection);
+        assertThat(attemptCollection.metricValues(CoreMetric.HTTP_REQUEST_ROUND_TRIP_TIME).get(0))
+            .isGreaterThanOrEqualTo(FIXED_DELAY);
     }
 
     @Test
-    public void apiCall_operationFailed_addsMetrics() {
+    public void apiCall_allRetryAttemptsFailedOf500() {
         stubErrorResponse();
-        assertThatThrownBy(() -> client.allTypes().join()).hasCauseInstanceOf(ProtocolRestJsonException.class);
+        assertThatThrownBy(() -> client.allTypes().join()).hasCauseInstanceOf(EmptyModeledException.class);
+
+        ArgumentCaptor<MetricCollection> collectionCaptor = ArgumentCaptor.forClass(MetricCollection.class);
+        verify(mockPublisher).publish(collectionCaptor.capture());
+
+        MetricCollection capturedCollection = collectionCaptor.getValue();
+        verifyApiCallCollection(capturedCollection);
+        assertThat(capturedCollection.children()).hasSize(MAX_RETRIES + 1);
+
+        capturedCollection.children().forEach(this::verifyFailedApiCallAttemptCollection);
+    }
+
+    private void verifyFailedApiCallAttemptCollection(MetricCollection requestMetrics) {
```

Comment:
Can you move this private method down to the other private methods?

Reply:
sure